### ISSUE TITLE
Behaviour with multi-gpu

### ISSUE THREAD

**Issue by ayushjain1144** [run-ai/runai#35]
Hi, 

Thank you for your awesome work!

Could you please let me know if run-ai natively handles multi-gpu training with DDP and would produce correct results? And if I have two GPUs, then `steps = 4` would still mean 4 sets of "effective" batches right? (i.e. with num_gpus=2, batch_size_per_gpu=2 and steps=4, the total batch size after gradient accumulation would be: 2*2*4=16?). Thank you so much!
